src/config.rs

```diff
@@ -87,12 +87,25 @@ pub struct Mouse {
     #[serde(deserialize_with = "deserialize_faux_scrollback_lines")]
     #[serde(default="default_faux_scrollback_lines")]
     pub faux_scrollback_lines: usize,
+
+    /// Swallow the first click to an unfocused window, or pass it through.
+    #[serde(deserialize_with = "failure_default")]
+    #[serde(default="default_ignore_unfocused_click")]
+    pub ignore_unfocused_click: bool,
 }
 
 fn default_faux_scrollback_lines() -> usize {
     1
 }
 
+fn default_ignore_unfocused_click() -> bool {
+    #[cfg(target_os="macos")]
+    let default_ignore_unfocused_click: bool = true;
+    #[cfg(not(target_os="macos"))]
+    let default_ignore_unfocused_click: bool = false;
+    default_ignore_unfocused_click
+}
```

Comment:
This looks a little difficult to read and is probably a lot more complicated than it needs to be.
Inside a method I'd probably recommend using the `if cfg!` rather than the annotations, this is a bit more like normal control flow of methods so it makes things a little simpler.

This should do the trick:
```rust
if cfg!(target_os="macos") {
    true
} else {
    false
}
````

Alternatively it would also be possible to add two separate methods and then use the annotations, but I think the macro should work just fine for this.

Reply:
You're right, that is simpler, thanks.